daemon/containerd/image_pull.go

```diff
@@ -68,6 +70,22 @@ func (i *ImageService) PullImage(ctx context.Context, image, tagOrDigest string,
 	infoHandler := snapshotters.AppendInfoHandlerWrapper(ref.String())
 	opts = append(opts, containerd.WithImageHandlerWrapper(infoHandler))
 
-	_, err = i.client.Pull(ctx, ref.String(), opts...)
-	return err
+	img, err := i.client.Pull(ctx, ref.String(), opts...)
+	if err != nil {
+		return err
+	}
+
+	logger := logrus.WithFields(logrus.Fields{
+		"digest": img.Target().Digest,
+		"remote": ref.String(),
+	})
+	logger.Info("image pulled")
+
+	// The pull succeeded, so try to remove any dangling image we have for this target
+	if err := i.client.ImageService().Delete(context.Background(), danglingImageName(img.Target().Digest)); err != nil {
+		if !cerrdefs.IsNotFound(err) {
+			logger.WithError(err).Warn("unexpected error when deleting dangling image")
```

Comment:
Oh! Silly question; does the `err` contain any reference to the "darling image name"? Because we set up the `logger` to have the digest and name for the image we pulled, and here we're deleting the dangling image 🤔 

Oh, I guess `digest` is "close" (so mostly a nit I guess)

Reply:
(If it's confusing, then we could inline the `WithFields()` instead of re-using a `logger`  `logger.WithFields(....).Warn("unexpected error when deleting dangling image")`)

Reply:
Maybe we could resolve the confusion with a slightly reworded error?

```suggestion
			logger.WithError(err).Warn("unexpected error while removing outdated dangling image reference")
```

Reply:
Nice! Thanks @tianon, sounds good :)